ambassador/ambassador/ir/irhttpmappinggroup.py

```diff
@@ -208,14 +208,22 @@ def add_cluster_for_mapping(ir: 'IR', aconf: Config, mapping: IRBaseMapping,
         # ...and return it. Done.
         return stored
 
-    def finalize(self, ir: 'IR', aconf: Config) -> List[IRCluster]:
+
+    def finalize(self, ir: 'IR', aconf: Config) -> bool:
+        if len(self.mappings) == 0:
+            errstr = "rejecting mapping group with no non-shadow mappings"
+            err = RichStatus.fromError(errstr)
+            aconf.post_error(err, self)
+            self.ir.logger.debug(errstr)
+            return False
+
         """
         Finalize a MappingGroup based on the attributes of its Mappings. Core elements get lifted into
         the Group so we can more easily build Envoy routes; host-redirect and shadow get handled, etc.
 
         :param ir: the IR we're working from
         :param aconf: the Config we're working from
-        :return: a list of the IRClusters this Group uses
```

Comment:
it looked like the caller wasn't using this return value, so I felt OK changing it